grudge/reductions.py

```diff
@@ -317,67 +327,121 @@ def elementwise_prg():
         )
 
 
-def elementwise_sum(dcoll: DiscretizationCollection, *args) -> DOFArray:
+def elementwise_sum(
```

Comment:
I think there's an opportunity to document the elementwise reductions more precisely: namely, mention that they return `DOFArrays` with the same number of degrees of freedom as the input, all set to the same value. Now that I type that out, that's kind of dumb, and kind of unnecessary when proper broadcasting is available. But that's for another time. -> #184 

The doc concern stands though, even in view of that direction.

Reply:
I've added notes to all the `elementwise_x` functions: https://github.com/inducer/grudge/pull/154/commits/e7478d4abbbcd540936c416004573b2907cfa347